File: src/main/java/com/tomatokey/framework/exception/ErrorResponse.java
package com.tomatokey.framework.exception;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import org.springframework.http.HttpStatus;

import java.io.Serializable;
import java.util.List;

@Getter
@Builder
@AllArgsConstructor
public class ErrorResponse implements Serializable {

    private int status;
    private String message;
    private List<ValidError> errors;

    public ErrorResponse(HttpStatus status, String message) {
        this.status = status.value();
        this.message = message;
    }

    public ErrorResponse(HttpStatus status, String message, List<ValidError> errors) {
        this.status = status.value();
        this.message = message;
        this.errors = errors;
    }

}

File: src/main/kotlin/com/prototype/framework/exception/ErrorResponse.kt
package com.prototype.framework.exception

import org.springframework.http.HttpStatus
import java.io.Serializable

data class ErrorResponse(
        val status: Int,
        val message: String,
        val errors: List<ValidError>? = null
) : Serializable {

    constructor(status: HttpStatus, message: String) : this(status.value(), message)
    constructor(status: HttpStatus, message: String, errors: List<ValidError>?) : this(status.value(), message, errors)

}